lib/livebook_web/live/hub_live/fly_component.ex

```diff
@@ -192,22 +219,49 @@ defmodule LivebookWeb.HubLive.FlyComponent do
         ]
       end
 
-    Enum.reverse(options ++ [disabled_option])
+    [disabled_option] ++ options
   end
 
-  defp create_fly(socket, params) do
-    if Hubs.hub_exists?(socket.assigns.selected_app.id) do
-      send(self(), {:flash_error, "Application already exists"})
-      {:noreply, assign(socket, data: params)}
-    else
-      save_fly(socket, params)
+  defp save_fly(socket, :new, params) do
+    case Fly.create_fly(socket.assigns.selected_app, params) do
+      {:ok, fly} ->
+        changeset =
+          fly
+          |> Fly.changeset(params)
+          |> put_action()
+
+        send(self(), {:flash, :info, "Hub created successfully"})
+        assign(socket, changeset: changeset, selected_app: fly, valid?: false)
+
+      {:error, changeset} ->
+        send(self(), {:flash, :error, "Failed to create Hub"})
+        assign(socket, changeset: put_action(changeset), valid?: changeset.valid?)
     end
   end
 
-  defp save_fly(socket, params) do
-    Fly.save_fly(socket.assigns.selected_app, params)
-    opts = select_options(socket.assigns.apps, params["application_id"])
+  defp save_fly(socket, :edit, params) do
+    case Fly.update_fly(socket.assigns.selected_app, params) do
+      {:ok, fly} ->
+        changeset =
+          fly
+          |> Fly.changeset(params)
+          |> put_action()
+
+        send(self(), {:flash, :info, "Hub updated successfully"})
+        assign(socket, changeset: changeset, selected_app: fly, valid?: changeset.valid?)
 
-    {:noreply, assign(socket, data: params, select_options: opts)}
+      {:error, changeset} ->
+        send(self(), {:flash, :error, "Failed to update Hub"})
+        assign(socket, changeset: put_action(changeset), valid?: changeset.valid?)
+    end
   end
+
+  defp clean_errors(changeset), do: Map.put(changeset, :errors, [])
+  defp put_action(changeset, action \\ :validate), do: Map.put(changeset, :action, action)
+
+  defp hub_color(changeset), do: get_field(changeset, :hub_color)
+  defp access_token(changeset), do: get_field(changeset, :access_token)
+
+  defdelegate get_field(changeset, field), to: Ecto.Changeset
+  defdelegate add_error(changeset, field, message), to: Ecto.Changeset
```

Comment:
Don't use delegates if those functions are not meant to be part of your API. Instead, you can import them!

Reply:
But in this case, it is probably just easier to use the full qualified name: `Ecto.Changeset.get_field` and so on.